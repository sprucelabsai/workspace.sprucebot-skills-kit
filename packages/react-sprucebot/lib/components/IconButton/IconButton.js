'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Icon = require('../Icon/Icon');

var _Icon2 = _interopRequireDefault(_Icon);

var _Button = require('../Button/Button');

var _Button2 = _interopRequireDefault(_Button);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var IconButton = function IconButton(_ref) {
	var children = _ref.children,
	    className = _ref.className,
	    loaderStyle = _ref.loaderStyle,
	    props = _objectWithoutProperties(_ref, ['children', 'className', 'loaderStyle']);

	return _react2.default.createElement(
		_Button2.default,
		_extends({
			loaderDark: true,
			loaderStyle: _extends({
				fontSize: '.5em'
			}, loaderStyle)
		}, props, {
			className: 'IconButton icon__button ' + (className || '')
		}),
		_react2.default.createElement(
			_Icon2.default,
			{ fontSize: props.fontSize },
			children
		)
	);
};

IconButton.propTypes = {
	onClick: _propTypes2.default.func,
	children: _propTypes2.default.any
};

exports.default = IconButton;