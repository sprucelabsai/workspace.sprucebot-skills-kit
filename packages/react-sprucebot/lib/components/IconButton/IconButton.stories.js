'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonInfo = require('@storybook/addon-info');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _IconButton = require('./IconButton');

var _IconButton2 = _interopRequireDefault(_IconButton);

var _IconButton3 = require('./IconButton.md');

var _IconButton4 = _interopRequireDefault(_IconButton3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('IconButton', module);
stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_IconButton4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_IconButton2.default,
		{ className: 'edit' },
		'edit'
	);
})));