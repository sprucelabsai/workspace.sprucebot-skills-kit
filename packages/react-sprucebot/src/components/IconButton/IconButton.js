import styled from 'styled-components'
import PropTypes from 'prop-types'
import React from 'react'

import Icon from '../Icon/Icon'

const Button = styled.button.attrs({
	className: ({ className }) => `IconButton icon-button ${className || ''}`
})`
	align-items: center;
	background: none;
	color: #00aac7;
	display: flex;
	font-size: 2em;
	margin: 0;
	padding: 0;
	width: auto;

	&:hover {
		background: none;
	}
`

const IconButton = ({ onClick, children, ...props }) => (
	<Button onClick={onClick} {...props}>
		<Icon>{children}</Icon>
	</Button>
)

IconButton.propTypes = {
	onClick: PropTypes.func,
	children: PropTypes.any
}

export default IconButton
