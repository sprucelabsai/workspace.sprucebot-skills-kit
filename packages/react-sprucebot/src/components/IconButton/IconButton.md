# Icon Button

Use this button for any interaction that can be visually communicated without text
