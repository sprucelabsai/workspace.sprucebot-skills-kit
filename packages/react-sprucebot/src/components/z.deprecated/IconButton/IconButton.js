import PropTypes from 'prop-types'
import React from 'react'
import Icon from '../Icon/Icon'
import Button from '../Button/Button'

const IconButton = ({ children, className, loaderStyle, ...props }) => (
	<Button
		loaderDark
		loaderStyle={{
			fontSize: '.5em',
			...loaderStyle
		}}
		{...props}
		className={`IconButton icon__button ${className || ''}`}
	>
		<Icon fontSize={props.fontSize}>{children}</Icon>
	</Button>
)

IconButton.propTypes = {
	onClick: PropTypes.func,
	children: PropTypes.any
}

export default IconButton
