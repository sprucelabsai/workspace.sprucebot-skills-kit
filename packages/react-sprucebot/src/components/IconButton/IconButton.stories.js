import React from 'react'
import { storiesOf } from '@storybook/react'
import { withInfo } from '@storybook/addon-info'
import { withKnobs } from '@storybook/addon-knobs/react'
import { withReadme } from 'storybook-readme'
import IconButton from './IconButton'
import readme from './IconButton.md'

const stories = storiesOf('IconButton', module)
stories.addDecorator(withKnobs)

stories.add('Interactive', withReadme(readme, withInfo()(() => <IconButton />)))
