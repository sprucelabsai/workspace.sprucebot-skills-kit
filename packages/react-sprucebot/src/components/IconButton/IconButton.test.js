import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import IconButton from './IconButton'

test('it renders', () => {
	const tree = renderer.create(<IconButton />).toJSON()
	expect(tree).toMatchSnapshot()
})
