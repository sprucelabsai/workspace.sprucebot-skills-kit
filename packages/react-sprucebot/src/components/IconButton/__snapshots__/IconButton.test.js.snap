// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton tests Should match the snapshot 1`] = `
<Button
  alt={false}
  busy={false}
  className="IconButton icon__button "
  loaderDark={true}
  loaderStyle={
    Object {
      "fontSize": ".5em",
    }
  }
  primary={false}
  remove={false}
  secondary={false}
  tag="button"
  toggle={false}
  type="button"
>
  <Icon />
</Button>
`;
